src/cache.js

```diff
@@ -33,12 +33,12 @@ var p = Cache.prototype
 
 p.put = function (key, value) {
   var removed
-  if (this.size === this.limit) {
-    removed = this.shift()
-  }
 
   var entry = this.get(key, true)
   if (!entry) {
+    if (this.size === this.limit) {
+      removed = this.shift()
+    }
```

Comment:
I the outer if could be avoided if we moved the inner after the entry was added/updated.

then:
* if entry did alread exist, size will still be the same
* if entry is a new one, size will be over the limit, and we can remove the head.